packages/nfts/contracts/trailblazers-season-2/BadgeRecruitmentV2.sol

```diff
@@ -0,0 +1,129 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.24;
+
+import "./BadgeRecruitment.sol";
+
+contract BadgeRecruitmentV2 is BadgeRecruitment {
+    /// @notice Events
+    event RecruitmentReset(
+        uint256 indexed cycleId, address indexed user, uint256 indexed s1TokenId, uint256 s1BadgeId
+    );
+
+    /// @notice Errors
+    error RECRUITMENT_ALREADY_COMPLETED();
+    error RECRUITMENT_NOT_FOUND();
+
+    /// @notice Updated version function
+    function version() external pure virtual returns (string memory) {
+        return "V2";
+    }
+
+    /// @notice Start a recruitment for a badge
+    /// @param _s1BadgeId The badge ID (s1)
+    /// @dev Not all badges are eligible for recruitment at the same time
+    /// @dev Defines a cooldown for the recruitment to be complete
+    /// @dev the cooldown is lesser the higher the Pass Tier
+    /// @dev Must be called from the s1 badges contract
+    function startRecruitment(
+        address _user,
+        uint256 _s1BadgeId,
+        uint256 _s1TokenId
+    )
+        external
+        virtual
+        onlyRole(S1_BADGES_ROLE)
+        recruitmentOpen(_s1BadgeId)
+        isNotMigrating(_user)
+        hasntMigratedInCycle(_s1BadgeId, _user, RecruitmentType.Migration)
+    {
```

Comment:
we also wanted an additional check to prevent starting a recruitment, when the duration would exceed the time left of the cycle